ledger/acctupdates.go

```diff
@@ -1036,35 +1039,54 @@ func (au *accountUpdates) deleteStoredCatchpoints(ctx context.Context, dbQueries
 	return nil
 }
 
-func (au *accountUpdates) accountsUpdateBalances(accountsDeltas map[basics.Address]accountDelta) (err error) {
+// accountsUpdateBalances applies the given deltas array to the merkle trie
+func (au *accountUpdates) accountsUpdateBalances(accountsDeltasRound []map[basics.Address]accountDelta, offset uint64) (err error) {
 	if au.catchpointInterval == 0 {
 		return nil
 	}
 	var added, deleted bool
-	for addr, delta := range accountsDeltas {
-		if !delta.old.IsZero() {
-			deleteHash := accountHashBuilder(addr, delta.old, protocol.Encode(&delta.old))
-			deleted, err = au.balancesTrie.Delete(deleteHash)
-			if err != nil {
-				return err
+	accumulatedChanges := 0
+	for i := uint64(0); i < offset; i++ {
+		accountsDeltas := accountsDeltasRound[i]
+		for addr, delta := range accountsDeltas {
+			if !delta.old.IsZero() {
+				deleteHash := accountHashBuilder(addr, delta.old, protocol.Encode(&delta.old))
+				deleted, err = au.balancesTrie.Delete(deleteHash)
+				if err != nil {
+					return err
+				}
+				if !deleted {
+					au.log.Warnf("failed to delete hash '%v' from merkle trie", deleteHash)
+				} else {
+					accumulatedChanges++
+				}
+
 			}
-			if !deleted {
-				au.log.Warnf("failed to delete hash '%v' from merkle trie", deleteHash)
+			if !delta.new.IsZero() {
+				addHash := accountHashBuilder(addr, delta.new, protocol.Encode(&delta.new))
+				added, err = au.balancesTrie.Add(addHash)
+				if err != nil {
+					return err
```

Comment:
is it OK to lose pending changes if at some point error happened? Maybe
```
defer func() {
	if accumulatedChanges > 0 {
		err = au.balancesTrie.Commit()
	}
}()
```
?

Reply:
yes, it's ok to loose changes if error occurs - 
in case of an error, the entire transaction would get rolled back, which includes all the pending operations.
The call to "au.balancesTrie.Commit()" would trickle down to a series of calls to `merkleCommitter.StorePage`, which uses the current transaction.

The reason we don't want to hold this off for too long is because it would create higher memory pressure for pending items. ( which in turn, make the single "Commit" call take a long time )

The number I've picked ( 256 ) is likely to be on the low end, and we probably could go higher.